internal/rpcserver/rpcserverhandlers_test.go

```diff
@@ -4388,6 +4551,561 @@ func TestHandleVerifyChain(t *testing.T) {
+				connMgr := defaultMockConnManager()
+				connMgr.connectedCount = 1
+				return connMgr
+			}(),
+			mockChainParams: func() *chaincfg.Params {
+				return chaincfg.SimNetParams()
+			}(),
+			mockMiningState: func() *testMiningState {
+				ms := defaultMockMiningState()
+				ms.miningAddrs = []dcrutil.Address{miningaddr}
+				ms.allowUnsyncedMining = false
+
+				tmplKey := getWorkTemplateKey(&block.Header)
+				workState := newWorkState()
+				workState.templatePool[tmplKey] = &block
+				ms.workState = workState
+
+				return ms
+			}(),
+			mockSyncManager: func() *testSyncManager {
+				syncManager := defaultMockSyncManager()
+				syncManager.isOrphan = true
+				return syncManager
+			}(),
+			result: false,
+		}, {
+			name:    "handleGetWork: submission ok",
+			handler: handleGetWork,
+			cmd: &types.GetWorkCmd{
+				Data: &submission,
+			},
+			mockConnManager: func() *testConnManager {
+				connMgr := defaultMockConnManager()
+				connMgr.connectedCount = 1
+				return connMgr
+			}(),
+			mockChainParams: func() *chaincfg.Params {
+				return chaincfg.SimNetParams()
+			}(),
+			mockMiningState: func() *testMiningState {
+				ms := defaultMockMiningState()
+				ms.miningAddrs = []dcrutil.Address{miningaddr}
+				ms.allowUnsyncedMining = false
+
+				tmplKey := getWorkTemplateKey(&block.Header)
+				workState := newWorkState()
+				workState.templatePool[tmplKey] = &block
+				ms.workState = workState
+
+				return ms
+			}(),
+			result: true,
+		}, {
+			name:    "handleGetWork: unknown parent for submission",
+			handler: handleGetWork,
+			cmd: &types.GetWorkCmd{
+				Data: &submission,
+			},
+			mockConnManager: func() *testConnManager {
+				connMgr := defaultMockConnManager()
+				connMgr.connectedCount = 1
+				return connMgr
+			}(),
+			mockChainParams: func() *chaincfg.Params {
+				return chaincfg.SimNetParams()
+			}(),
+			mockMiningState: func() *testMiningState {
+				ms := defaultMockMiningState()
+				ms.miningAddrs = []dcrutil.Address{miningaddr}
+				ms.allowUnsyncedMining = false
+
+				tmplKey := getWorkTemplateKey(&block.Header)
+				workState := newWorkState()
+				workState.templatePool[tmplKey] = &block
+				ms.workState = workState
+
+				return ms
+			}(),
+			mockSyncManager: func() *testSyncManager {
+				syncManager := defaultMockSyncManager()
+				syncManager.isOrphan = false
+				syncManager.submitBlockErr = errors.New("unknown parent")
+				return syncManager
+			}(),
+			wantErr: true,
+			errCode: dcrjson.ErrRPCInternal.Code,
+		},
+		{
+			name:    "handleGetWork: invalid hex",
+			handler: handleGetWork,
+			cmd: &types.GetWorkCmd{
+				Data: &invalidHexSub,
+			},
+			mockConnManager: func() *testConnManager {
+				connMgr := defaultMockConnManager()
+				connMgr.connectedCount = 1
+				return connMgr
+			}(),
+			mockChainParams: func() *chaincfg.Params {
+				return chaincfg.SimNetParams()
+			}(),
+			mockMiningState: func() *testMiningState {
```

Comment:
This is identically repeated a bunch.  It would save a lot of code to define it once and use the local instance.